deployer/src/main/java/org/apache/brooklyn/entity/cloudfoundry/webapp/java/JavaCloudFoundryPaasWebAppImpl.java

```diff
@@ -63,4 +81,147 @@ public String getBuildpack() {
+    }
+
+    @Override
+    protected void connectSensors() {
+        super.connectSensors();
+
+        String managementUri = String.format("%s/monitoring?format=json",
+                getAttribute(CloudFoundryWebApp.ROOT_URL));
+
+        setAttribute(JavaCloudFoundryPaasWebApp.MONITOR_URL, managementUri);
+        String monitorResource = this.getConfig(MAIN_MONITOR_RESOURCE);
+
+        httpFeed = HttpFeed.builder()
+                .entity(this)
+                .period(200)
+                .baseUri(managementUri)
+                .poll(new HttpPollConfig<Long>(USED_MEMORY)
+                        .checkSuccess(HttpValueFunctions.responseCodeEquals(200))
+                        .onSuccess(HttpValueFunctions.<Long>jsonContentsFromPath("$.list[12].memoryInformations.usedMemory")))
+                .poll(new HttpPollConfig<Double>(DURATION_SUM).onSuccess(
+                        Functionals.chain(
+                                HttpValueFunctions.stringContentsFunction(),
+                                findAMonitorData(monitorResource, "durationsSum"))))
+                .poll(new HttpPollConfig<Double>(RESOURCE_HITS).onSuccess(
+                        Functionals.chain(
+                                HttpValueFunctions.stringContentsFunction(),
+                                findAMonitorData(monitorResource, "hits"))))
+                .poll(new HttpPollConfig<Double>(SERVER_PROCESSING_TIME)
+                        .checkSuccess(HttpValueFunctions.responseCodeEquals(200))
+                        .onSuccess(HttpValueFunctions.<Double>jsonContentsFromPath("$.list[12].tomcatInformationsList[0].processingTime")))
+                .poll(new HttpPollConfig<Double>(SERVER_REQUESTS)
+                        .checkSuccess(HttpValueFunctions.responseCodeEquals(200))
+                        .onSuccess(HttpValueFunctions.<Double>jsonContentsFromPath("$.list[12].tomcatInformationsList[0].requestCount")))
+                .build();
+
+        connectServiceLatencySensor();
+        connectResourceLatencySensor();
+        connectServiceRequestPerSecondSensor();
+    }
+
+    private void connectServiceLatencySensor() {
+        serverLatencyFeed = FunctionFeed.builder()
+                .entity(this)
+                .period(Duration.seconds(2))
+                .poll(new FunctionPollConfig<Double, Double>(SERVER_LATENCY)
+                        .onException(Functions.constant(0.0))
+                        .callable(new Callable<Double>() {
+                            public Double call() {
+
+                                Double total = getAttribute(SERVER_PROCESSING_TIME);
+                                Double num = getAttribute(SERVER_REQUESTS);
+                                return total / num;
+                            }
+                        }))
+                .build();
+    }
+
+    private void connectServiceRequestPerSecondSensor() {
+        serverLatencyFeed = FunctionFeed.builder()
```

Comment:
You are overwriting here the value of serverLatencyFeed calculated in connectServiceLatencySensor(). Why?